finagle-mysql/src/main/scala/com/twitter/finagle/mysql/CanBeParameter.scala

```diff
@@ -0,0 +1,182 @@
+package com.twitter.finagle.exp.mysql
+
+import com.twitter.finagle.exp.mysql.transport.{Buffer, BufferWriter}
+
+trait CanBeParameter[-A] { outer =>
+  /**
+   * Returns the size of the given parameter in its MySQL binary representation.
+   **/
+  def sizeOf(param: A): Int
+
+  /**
+   * Retrieves the MySQL type code for the given parameter.
+   * */
+  def typeCode(param: A): Short
```

Comment:
It probably wasn't the best idea to derive the blob type from the size of the array, but we should revisit this separately. I think retaining current behavior is a good idea!

@mosesn The blobs are length encoded, but they still need to match your schema. I suspect mysql will complain otherwise.